orchagent/fgnhgorch.cpp

```diff
@@ -820,11 +863,33 @@ bool FgNhgOrch::setNewNhgMembers(FGNextHopGroupEntry &syncd_fg_route_entry, FgNh
             return false;
         }
 
+        // recover state before warm reboot
+        if (nexthopsMap != m_recoveryMap.end())
+        {
+            is_warm_reboot = true;
```

Comment:
>is_warm_reboot [](start = 12, length = 14)

Is it better to check m_recoveryMap.empty()? If no, then it is warm-reboot.

Reply:
auto nexthopsMap = m_recoveryMap.find(ipPrefix.to_string()); this checks if the route-addition is required to be recover after warm reboot. so checking if the route prefix can be found in m_recoveryMap. 
I think we should check the prefix existing instead of only recoverymap is empty. what do you think?